### ISSUE TITLE
Should enhance functions take a map and return a map?

### ISSUE THREAD

**Issue by hayesla** [sunpy/sunkit-image#189]
### Describe the feature

At the moment, `sunkit_image.enhance.mgn` takes data and then returns data. A user has to pass extract the data from a map, apply the function, then reconstruct a map, wouldnt it make more sense that this (and then any other function in sunkit_image.enhance` take and return a map?

**Comment by Ghaithq** [comment]
Isn't this the case? It is stated in the code documentation that it does return a map depending on the input.
```
        Normalized image. If a map is input, a map is returned with new data
        and the same metadata```

**Comment by wtbarnes** [comment]
Yeah this was closed a while back by #135.

**Comment by wtbarnes** [comment]
I'm closing this as I'm pretty sure this is already possible but feel free to reopen if for some reason I'm missing something!
